onchain_chainalysis.go

```diff
@@ -0,0 +1,40 @@
+package chainalysis
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/0xsequence/ethkit/ethrpc"
+)
+
+type onChainAlysis struct {
+	provider *ethrpc.Provider
+}
+
+func NewOnChainAlysis() (ChainAlysis, error) {
+	provider, err := ethrpc.NewProvider("https://nodes.sequence.app/mainnet")
+	if err != nil {
+		return nil, err
+	}
+	return &onChainAlysis{
+		provider: provider,
+	}, nil
+}
+
+func (o *onChainAlysis) IsSanctioned(address string) (bool, error) {
+	ctx, _ := context.WithDeadline(context.Background(), time.Now().Add(5*time.Second))
```

Comment:
we want to avoid using this, as it'll be too intensive on the node, this is why we have the index.

You can keep this method, but have it read from the in-memory index which gets updated behind the scenes every 10minutes, etc..